grpc/interop/src/main/scala/grpc/testing/ServerMain.scala

```diff
@@ -0,0 +1,92 @@
+package grpc.testing
+
+import com.google.protobuf.CodedOutputStream
+import com.twitter.conversions.time._
+import com.twitter.finagle.buoyant.H2
+import com.twitter.io.Buf
+import com.twitter.util.{Await, Future, Promise, Throw}
+import io.buoyant.grpc.runtime.{Stream, ServerDispatcher}
+import java.nio.ByteBuffer
+import java.util.Arrays
+
+/**
+ * The main object for running a gRPC interop server.
+ *
+ * The interop tests are described here:
+ *   https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md
+ */
+object ServerMain {
+  def main(args: Array[String]) {
+
+    val iface = new TestService {
+      def emptyCall(empty: Empty): Future[Empty] = {
+        Future.value(new Empty())
+      }
+
+      def unaryCall(req: SimpleRequest): Future[SimpleResponse] = {
+        Future.value(new SimpleResponse(None, None, None))
+      }
+
+      def cacheableUnaryCall(req: SimpleRequest): Future[SimpleResponse] = {
+        Future.value(new SimpleResponse(None, None, None))
+      }
+
+      /**
+       * Echo back each request with a Payload having the requested size
+       */
+      def fullDuplexCall(req: Stream[StreamingOutputCallRequest]): Stream[StreamingOutputCallResponse] = {
+        val tx = Stream[StreamingOutputCallResponse]
+
+        req.recv().map { outputCallRequest =>
+          outputCallRequest.value.payload.map { payload =>
+            payload.body.map { body =>
+              val chars = Array.fill(body.length) { 0.toByte } // {} specializes us to a Byte array
+              val toSend = Buf.ByteArray.Owned(chars)
+              tx.send(StreamingOutputCallResponse(Some(Payload(None, Some(toSend)))))
+            }
+          }
+        }
```

Comment:
This only requests a single frame from the stream.  I think this should be something like:
```scala
      def fullDuplexCall(reqs: Stream[StreamingOutputCallRequest]): Stream[StreamingOutputCallResponse] = {
        val rsps = Stream[StreamingOutputCallResponse]
        def process(): Future[Unit] =
          reqs.recv().respond {
            case Throw(e) => rsps.close()
            case Return(req) =>
              req.value.payload.flatMap(_.body) match {
                case None => rsps.close()
                case Some(body) =>
                  val chars = Array.fill(body.length) { 0.toByte } // {} specializes us to a Byte array
                  val toSend = Buf.ByteArray.Owned(chars)
                  val msg = StreamingOutputCallResponse(Some(Payload(None, Some(toSend))))
                  rsps.send(msg).before(process())
              }
          }
        process()
        rsps
      }
```